### PR TITLE
[WIP] Feat/threaded simulation

### PR DESCRIPTION
**Pull request by djrtwo**
Initial stab at a threaded simulator.

Tests will def fail. Needs a lot of cleanup and documentation.

can run with following command or similar:
`python casper.py rand --protocol blockchain --validators 5 --rounds 100 --report-interval 1 --hide-display --save True --network linear --simulator multi-threaded`

### CODE REVIEW

File: simulations/threaded_simulation.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            validator_client.should_make_new_message = msg_strategy
+
+        self.plot_tool = protocol.PlotTool(False, save, network.global_view, validator_set)
+
+    def run(self):
+        print("starting simulation")
+        self._send_initial_messages()
+
+        for validator_client in self.validator_clients:
+            validator_client.start()
```
Comment (SeanAvery) [inline simulations/threaded_simulation.py]: This is looking very dope!

Might be helpful to have a comment here that we are initializing threads and `start()` is apart of the Thread Class. Not  exactly sure how validator_client.run() gets hit by this but I see it is working :)

Comment (SeanAvery) [inline simulations/threaded_simulation.py]: This is looking very dope!

Might be helpful to have a comment here that we are initializing threads and start() is apart of the Thread Class. Not exactly sure how validator_client.run() gets hit by this but I see it is working :)

Comment (djrtwo) [reply simulations/threaded_simulation.py]: Will do!

Objects that inherit from Thread get this `start` method. `start` initializes things and sets the thread in motion, running the overidden `run` method.

**Comment by djrtwo** [comment]
merging into #158

**Pull request merged**